cmd/register/files.go

```diff
@@ -0,0 +1,179 @@
+package register
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/golang/protobuf/proto"
+	"github.com/lyft/flytectl/cmd/config"
+	cmdCore "github.com/lyft/flytectl/cmd/core"
+	"github.com/lyft/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/lyft/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/lyft/flytestdlib/logger"
+	"io/ioutil"
+	"sort"
+	"strings"
+)
+
+const identifierFileSuffix = "identifier"
+
+func unMarshalContents(ctx context.Context, fileContents []byte, fname string) (proto.Message, error) {
+	workflowSpec := &admin.WorkflowSpec{}
+	if err := proto.Unmarshal(fileContents, workflowSpec); err == nil {
+		return workflowSpec, nil
+	}
+	logger.Debugf(ctx, "Failed to unmarshal file %v for workflow type", fname)
+	taskSpec := &admin.TaskSpec{}
+	if err := proto.Unmarshal(fileContents, taskSpec); err == nil {
+		return taskSpec, nil
+	}
+	logger.Debugf(ctx, "Failed to unmarshal  file %v for task type", fname)
+	launchPlanSpec := &admin.LaunchPlanSpec{}
+	if err := proto.Unmarshal(fileContents, launchPlanSpec); err == nil {
+		return launchPlanSpec, nil
+	}
+	logger.Debugf(ctx, "Failed to unmarshal file %v for launch plan type", fname)
+	return nil, errors.New(fmt.Sprintf("Failed unmarshalling file %v", fname))
+
+}
+
+func register(ctx context.Context, message proto.Message, name string, cmdCtx cmdCore.CommandContext) error {
+	switch message.(type) {
+	case *admin.LaunchPlanSpec:
+		_, err := cmdCtx.AdminClient().CreateLaunchPlan(ctx, &admin.LaunchPlanCreateRequest{
+			Id: &core.Identifier{
+				ResourceType: core.ResourceType_LAUNCH_PLAN,
+				Project:      config.GetConfig().Project,
+				Domain:       config.GetConfig().Domain,
+				Name:         name,
+				Version:      GetConfig().version,
+			},
+			Spec: message.(*admin.LaunchPlanSpec),
+		})
+		return err
+	case *admin.WorkflowSpec:
+		_, err := cmdCtx.AdminClient().CreateWorkflow(ctx, &admin.WorkflowCreateRequest{
+			Id: &core.Identifier{
+				ResourceType: core.ResourceType_WORKFLOW,
+				Project:      config.GetConfig().Project,
+				Domain:       config.GetConfig().Domain,
+				Name:         name,
+				Version:      GetConfig().version,
+			},
+			Spec: message.(*admin.WorkflowSpec),
+		})
+		return err
+	case *admin.TaskSpec:
+		_, err := cmdCtx.AdminClient().CreateTask(ctx, &admin.TaskCreateRequest{
+			Id: &core.Identifier{
+				ResourceType: core.ResourceType_TASK,
+				Project:      config.GetConfig().Project,
+				Domain:       config.GetConfig().Domain,
+				Name:         name,
+				Version:      GetConfig().version,
+			},
+			Spec: message.(*admin.TaskSpec),
+		})
+		return err
+	default:
+		return errors.New(fmt.Sprintf("Failed registering unknown entity  %v type as part of %v file", message, name))
+	}
+}
+
+func hydrateNode(node *core.Node) {
+	targetNode := node.Target
+	switch targetNode.(type) {
+	case *core.Node_TaskNode:
```

Comment:
I think we need to fix this for BranchNode.

Reply:
can we add a default case, when none of the case statements match, error out

Reply:
Yes that in the list of pending items. Have added that in this MR itself now.